checker_test.go

```diff
@@ -970,6 +975,179 @@ got args:
 want args:
   regexp
 `,
+}, {
+	about:   "ErrorAs: exact match",
+	checker: qt.ErrorAs,
+	got:     error(errBadWolf),
+	args:    []interface{}{errTestPtr()},
+	expectedNegateFailure: `
+error:
+  unexpected success
+got:
+  bad wolf
+    file:line
+as:
+  &&quicktest_test.errTest{msg:"bad wolf", formatted:true}
+`,
+}, {
+	about:   "ErrorAs: wrapped match",
+	checker: qt.ErrorAs,
+	got:     fmt.Errorf("wrapped: %w", error(errBadWolf)),
+	args:    []interface{}{errTestPtr()},
+	expectedNegateFailure: `
+error:
+  unexpected success
+got:
+  e"wrapped: bad wolf\n  file:line"
+as:
+  &&quicktest_test.errTest{msg:"bad wolf", formatted:true}
+`,
+}, {
+	about:   "ErrorAs: fails if nil error",
+	checker: qt.ErrorAs,
+	got:     nil,
+	args:    []interface{}{errTestPtr()},
+	expectedCheckFailure: `
+error:
+  got nil error but want non-nil
+got:
+  nil
+as:
+  &(*quicktest_test.errTest)(nil)
+`,
+}, {
+	about:   "ErrorAs: fails if mismatch",
+	checker: qt.ErrorAs,
+	got:     errors.New("other error"),
+	args:    []interface{}{errTestPtr()},
+	expectedCheckFailure: `
+error:
+  error is not the expected error
+got:
+  e"other error"
+as:
+  &(*quicktest_test.errTest)(nil)
+`,
+}, {
+	about:   "ErrorAs: bad check if invalid error",
+	checker: qt.ErrorAs,
+	got:     "not an error",
+	args:    []interface{}{errTestPtr()},
+	expectedCheckFailure: `
+error:
+  bad check: first argument is not an error
+got:
+  "not an error"
+`,
+	expectedNegateFailure: `
+error:
+  bad check: first argument is not an error
+got:
+  "not an error"
+`,
+}, {
+	about:   "ErrorAs: bad check if invalid as",
+	checker: qt.ErrorAs,
+	got:     error(errBadWolf),
+	args:    []interface{}{&struct{}{}},
+	expectedCheckFailure: `
+error:
+  bad check: errors: *target must be interface or implement error
+`,
+	expectedNegateFailure: `
+error:
+  bad check: errors: *target must be interface or implement error
+`,
+}, {
+	about:   "ErrorIs: exact match",
+	checker: qt.ErrorIs,
+	got:     error(errBadWolf),
+	args:    []interface{}{errBadWolf},
+	expectedNegateFailure: `
+error:
+  unexpected success
+got:
+  bad wolf
+    file:line
+want:
+  <same as "got">
+`,
+}, {
+	about:   "ErrorIs: wrapped match",
+	checker: qt.ErrorIs,
+	got:     fmt.Errorf("wrapped: %w", error(errBadWolf)),
+	args:    []interface{}{errBadWolf},
+	expectedNegateFailure: `
+error:
+  unexpected success
+got:
+  e"wrapped: bad wolf\n  file:line"
+want:
+  bad wolf
+    file:line
+`,
+}, {
+	about:   "ErrorIs: fails if nil error",
+	checker: qt.ErrorIs,
+	got:     nil,
+	args:    []interface{}{errBadWolf},
+	expectedCheckFailure: `
+error:
+  got nil error but want non-nil
+got:
+  nil
+want:
+  bad wolf
+    file:line
+`,
+}, {
+	about:   "ErrorIs: fails if mismatch",
+	checker: qt.ErrorIs,
+	got:     error(errBadWolf),
+	args:    []interface{}{errors.New("other error")},
+	expectedCheckFailure: `
+error:
+  error is not the expected error
```

Comment:
`want error not found in got error chain`?

Reply:
:+1: